cli/server.go

```diff
@@ -814,11 +814,32 @@ func (r *RootCmd) Server(newAPI func(context.Context, *coderd.Options) (*coderd.
 				if err != nil {
 					return xerrors.Errorf("create telemetry reporter: %w", err)
 				}
+				go options.Telemetry.RunSnapshotter()
 				defer options.Telemetry.Close()
 			} else {
 				logger.Warn(ctx, fmt.Sprintf(`telemetry disabled, unable to notify of security issues. Read more: %s/admin/setup/telemetry`, vals.DocsURL.String()))
 			}
 
+			if !vals.Telemetry.Enable.Value() {
+				reporter, err := telemetry.New(telemetry.Options{
+					DeploymentID:         deploymentID,
+					Database:             options.Database,
+					Logger:               logger.Named("telemetry"),
+					URL:                  vals.Telemetry.URL.Value(),
+					DisableReportOnClose: true,
+				})
+				if err != nil {
+					logger.Debug(ctx, "create telemetry reporter (disabled)", slog.Error(err))
+				} else {
+					go func() {
+						defer reporter.Close()
+						if err := reporter.ReportDisabledIfNeeded(); err != nil {
+							logger.Debug(ctx, "failed to report disabled telemetry", slog.Error(err))
+						}
+					}()
+				}
+			}
+
```

Comment:
Should this not just be in the else block?


```suggestion
			} else {
				logger.Warn(ctx, fmt.Sprintf(`telemetry disabled, unable to notify of security issues. Read more: %s/admin/setup/telemetry`, vals.DocsURL.String()))
				reporter, err := telemetry.New(telemetry.Options{
					DeploymentID:         deploymentID,
					Database:             options.Database,
					Logger:               logger.Named("telemetry"),
					URL:                  vals.Telemetry.URL.Value(),
					DisableReportOnClose: true,
				})
				if err != nil {
					logger.Debug(ctx, "create telemetry reporter (disabled)", slog.Error(err))
				} else {
					go func() {
						defer reporter.Close()
						if err := reporter.ReportDisabledIfNeeded(); err != nil {
							logger.Debug(ctx, "failed to report disabled telemetry", slog.Error(err))
						}
					}()
				}
			}

```


---

Is there a way we can just configure `telemetry.New` and pass in the `enabled` flag into the telemetry struct? I ask because this init code is now ~60 lines long, wondering we can push the `else` condition just into the `RunSnapshotter`.

Just a thought.